### PR TITLE
[MRG] add finetuning for vision transformer models

### LINKED ISSUE #6: hydra for configing
https://github.com/facebookresearch/hydra

### PR DESCRIPTION
**Pull request by STALLAAAA**
1. add finetune for VisionTransformer, Inception3, DenseNet, SqueezeNet, VGG, AlexNet, ResNet
2. add hydra config and logger for finetune
close #6 #5 
Here are some logs of vit fine tuning on Places365:
7 epochs：
image
image
image
 14 epochs：
image

### CODE REVIEW

File: basic.yml [hunk 1]
```diff
@@ -1,18 +0,0 @@
-data_path: '/home/zhz/dataset/places365_standard'
```
Comment (HuaizhengZhang) [inline basic.yml:1]: I think this basic.yml should be kept? for training from the scratch

File: tools/finetune.py [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+
+    # Train and evaluate
+    model_ft = train_model(model, dataloaders_dict, device, criterion, optimizer_ft, logger,
+                                 print_freq=cfgs.print_freq, num_epochs=cfgs.epochs, is_inception=(cfgs.arch == "inception"))
+    mkdir(cfgs.weight_dir)
+    torch.save(model_ft.state_dict(), os.path.join(cfgs.weight_dir, cfgs.arch) + '.ckpt')
+    logger.info("model is saved at {}".format(os.path.abspath(os.path.join(cfgs.weight_dir, cfgs.arch) + '.ckpt')))
+
+
+def train_model(model, dataloaders, device, criterion, optimizer, logger, print_freq, num_epochs, is_inception=False):
```
Comment (HuaizhengZhang) [inline tools/finetune.py:105]: add a reference for this?

Comment (HuaizhengZhang) [reply tools/finetune.py:105]: the referece code link

File: utils/metric.py [hunk 3]
```diff
@@ -12,6 +12,6 @@ def accuracy(output, target, topk=(1,)):
 
         res = []
         for k in topk:
-            correct_k = correct[:k].view(-1).float().sum(0, keepdim=True)
+            correct_k = correct[:k].reshape(-1).float().sum(0, keepdim=True)
```
Comment (HuaizhengZhang) [inline utils/metric.py:15]: will this influence the previous results?

**Comment by HuaizhengZhang** [comment]
image
can not print the model structure as what I did before?

**Comment by HuaizhengZhang** [comment]
You can train more epochs according to the paper to check if the ViT is really good

And update the model_zoo by adding it

File: utils/metric.py (hunk 3 above)
Comment (STALLAAAA) [reply utils/metric.py:15]: I have compared this with torch metrics, and the results are the same. So I think this is just fine. BTW, this function will be replaced by `torch metrics` in the near future.

File: tools/finetune.py (hunk 2 above)
Comment (STALLAAAA) [reply tools/finetune.py:105]: I will add ref soon :https://pytorch.org/tutorials/beginner/finetuning_torchvision_models_tutorial.html

**Comment by STALLAAAA** [comment]
> image can not print the model structure as what I did before?

I will add it.

**Comment by STALLAAAA** [comment]
> You can train more epochs according to the paper to check if the ViT is really good
> 
> And update the model_zoo by adding it
@HuaizhengZhang accoding to http://arxiv.org/abs/2010.11929:
image
I fine tune vit on places365 for 7/14 epochs. ACC@1 rises from 0.52 to 0.54, ACC@5 rises from 0.83 to 0.85

**Review by HuaizhengZhang — APPROVED** [review]
